src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -13431,6 +13431,31 @@ private static void VerifyTypes(SemanticModel model, ExpressionSyntax expr, stri
             Assert.Equal(expectedConversionKind, conversion.Kind);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72541")]
+        public void NamedArgumentConversion()
+        {
+            var source = """
+                #nullable enable
+                using System.Collections.Generic;
+
+                static class C
+                {
+                    static void Main()
+                    {
+                        C.M(y: [new D { }]);
+                    }
+                    static void M(string x, IReadOnlyList<D> y) { }
+                }
+
+                class D { }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,11): error CS7036: There is no argument given that corresponds to the required parameter 'x' of 'C.M(string, IReadOnlyList<D>)'
+                //         C.M(y: [new D { }]);
+                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "M").WithArguments("x", "C.M(string, System.Collections.Generic.IReadOnlyList<D>)").WithLocation(8, 11));
+        }
```

Comment:
Is there a similar issue with target-typed collection initializers? For instance, with:
```csharp
#nullable enable
using System.Collections.Generic;

static class C
{
    static void Main()
    {
        C.M(y: new() { new D() { } });
    }
    static void M(string x, List<D> y) { }
}

class D { }
```

Reply:
Doesn't look like there is, but will add a test. Thanks.

Reply:
Yes, thanks. Only a similar assert fails, though, otherwise there's no error (so it's not observable in Release builds).